src/parser/classes/GridMovie.ts

```diff
@@ -0,0 +1,34 @@
+import Parser from '../index.js';
+import Text from './misc/Text.js';
+import Thumbnail from './misc/Thumbnail.js';
+import NavigationEndpoint from './NavigationEndpoint.js';
+import { YTNode } from '../helpers.js';
+import MetadataBadge from './MetadataBadge.js';
+
+class GridMovie extends YTNode {
+  static type = 'GridMovie';
+
+  id: string;
+  title: Text;
+  thumbnails: Thumbnail[];
+  duration: Text | null;
+  endpoint: NavigationEndpoint;
+  badges: MetadataBadge[];
+  metadata: Text;
+  thumbnail_overlays;
+
+  constructor(data: any) {
+    super();
+    const length_alt = data.thumbnailOverlays.find((overlay: any) => overlay.hasOwnProperty('thumbnailOverlayTimeStatusRenderer'))?.thumbnailOverlayTimeStatusRenderer;
```

Comment:
Please use the parsed thumbnail_overlays array (see bottom of this constructor) instead of the raw data (ideally the only place in YouTube.js that should be handling raw data and have any mention of renderers, is the Parser itself).

Here is an example of how you can do it:

https://github.com/LuanRT/YouTube.js/blob/main/src/parser/classes/PlaylistVideo.ts#L56

Reply:
I actually mirrored this part from `GridVideo`. 

Type of `duration` is `Text` which I prefer for consistency with `GridVideo`. If I obtain `length_alt` through parsed `thumb_overlays`, then I would have to do something like this for setting value of `duration`:

```
const length_alt = this.thumbnail_overlays.firstOfType(ThumbnailOverlayTimeStatus);
this.duration = data.lengthText ? new Text(data.lengthText) : length_alt?.text ? new Text({ simpleText: length_alt.text }) : null;
```
I'm fine with either way. What do you think?

Reply:
Yup, this is fine.